src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogWebhookBuildLogic.java

```diff
@@ -132,6 +132,25 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
         payload.put("unique_id", buildData.getBuildTag(""));
         payload.put("name", buildData.getBaseJobName(""));
 
+        // User
+        {
+            JSONObject userPayload = new JSONObject();
+            userPayload.put("name", buildData.getUserId());
+            if(StringUtils.isNotEmpty(buildData.getUserEmail(""))){
+                userPayload.put("email", buildData.getUserEmail(""));
+            }
+            payload.put("user", userPayload);
+        }
+
+        // Pipeline Parameters
+        if (!buildData.getBuildParameters().isEmpty()) {
+            JSONObject parametersPayload = new JSONObject();
+            for (Map.Entry<String, String> parameter : buildData.getBuildParameters().entrySet()) {
+                parametersPayload.put(parameter.getKey(), parameter.getValue());
```

Comment:
In the old version, where JSON was constructed manually, we used to do `StringEscapeUtils.escapeJavaScript(...)` on keys and values. Not sure what kind of data we have in build parameters. If this is something that comes from the user, we should escape it, I think.

Reply:
I think that's because we were creating the JSON by concatenating strings, so quotes and the like had to be escaped, but now that we use JSONObject it should already escape what it needs when serializing.